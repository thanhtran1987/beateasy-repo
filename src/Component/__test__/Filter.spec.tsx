import * as React from 'react';
import { shallow, mount } from 'enzyme';

import TestComponent from '../Filter';

describe('Filter', () => {
  it('should render correctly in default state', () => {
    const component = shallow(<TestComponent />);
    expect(component).toMatchSnapshot();
  });
  it('should dispatch correctly onclick for filter buttons', () => {
    const mockProps = {
      history: { push: jest.fn(), goBack: jest.fn() },
      setFilter: jest.fn(),
    };
    const component = mount(<TestComponent setFilter={mockProps.setFilter} filter={0} />);
    const target = component.find('button');
    target.at(0).simulate('click');
    target.at(1).simulate('click');
    target.at(2).simulate('click');
    target.at(3).simulate('click');
    expect(mockProps.setFilter).toHaveBeenCalledTimes(4);
  });
});
