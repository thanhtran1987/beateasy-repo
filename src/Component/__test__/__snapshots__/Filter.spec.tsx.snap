// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter should render correctly in default state 1`] = `
<styled.div>
  <styled.button
    className=""
    onClick={[Function]}
  >
    All
  </styled.button>
  <styled.button
    className=""
    onClick={[Function]}
  >
    <Component />
  </styled.button>
  <styled.button
    className=""
    onClick={[Function]}
  >
    <Component />
  </styled.button>
  <styled.button
    className=""
    onClick={[Function]}
  >
    <Component />
  </styled.button>
</styled.div>
`;
